wpilibc/src/main/native/cppcs/RobotBase.cpp

```diff
@@ -257,65 +217,34 @@ RobotBase::RobotBase() {
-                   HALUsageReporting::kDashboard_Glass);
-        m_dashboardDetected = true;
-      } else if (event.GetConnectionInfo()->remote_id.starts_with(
-                     "SmartDashboard")) {
-        HAL_Report(HALUsageReporting::kResourceType_Dashboard,
-                   HALUsageReporting::kDashboard_SmartDashboard);
-        m_dashboardDetected = true;
-      } else if (event.GetConnectionInfo()->remote_id.starts_with(
-                     "shuffleboard")) {
-        HAL_Report(HALUsageReporting::kResourceType_Dashboard,
-                   HALUsageReporting::kDashboard_Shuffleboard);
-        m_dashboardDetected = true;
-      } else if (event.GetConnectionInfo()->remote_id.starts_with("elastic") ||
-                 event.GetConnectionInfo()->remote_id.starts_with("Elastic")) {
-        HAL_Report(HALUsageReporting::kResourceType_Dashboard,
-                   HALUsageReporting::kDashboard_Elastic);
-        m_dashboardDetected = true;
-      } else if (event.GetConnectionInfo()->remote_id.starts_with(
-                     "Dashboard")) {
-        HAL_Report(HALUsageReporting::kResourceType_Dashboard,
-                   HALUsageReporting::kDashboard_LabVIEW);
-        m_dashboardDetected = true;
-      } else if (event.GetConnectionInfo()->remote_id.starts_with(
-                     "AdvantageScope")) {
-        HAL_Report(HALUsageReporting::kResourceType_Dashboard,
-                   HALUsageReporting::kDashboard_AdvantageScope);
-        m_dashboardDetected = true;
-      } else if (event.GetConnectionInfo()->remote_id.starts_with(
-                     "QFRCDashboard")) {
-        HAL_Report(HALUsageReporting::kResourceType_Dashboard,
-                   HALUsageReporting::kDashboard_QFRCDashboard);
-        m_dashboardDetected = true;
-      } else if (event.GetConnectionInfo()->remote_id.starts_with(
-                     "FRC Web Components")) {
-        HAL_Report(HALUsageReporting::kResourceType_Dashboard,
-                   HALUsageReporting::kDashboard_FRCWebComponents);
-        m_dashboardDetected = true;
-      } else {
-        if (!m_dashboardDetected) {
-          size_t delim = event.GetConnectionInfo()->remote_id.find('@');
-          if (delim != std::string::npos) {
-            HAL_Report(
-                HALUsageReporting::kResourceType_Dashboard,
-                HALUsageReporting::kDashboard_Unknown, 0,
-                event.GetConnectionInfo()->remote_id.substr(0, delim).c_str());
+  connListenerHandle =
+      inst.AddConnectionListener(false, [&](const nt::Event& event) {
+        if (event.Is(nt::EventFlags::kConnected)) {
+          auto connInfo = event.GetConnectionInfo();
+          if (connInfo->remote_id.starts_with("glass") ||
+              connInfo->remote_id.starts_with("SmartDashboard") ||
+              connInfo->remote_id.starts_with("shuffleboard") ||
+              connInfo->remote_id.starts_with("elastic") ||
+              connInfo->remote_id.starts_with("Elastic") ||
```

Comment:
Minor- Do we care about Elastic having two different names it might be reported as? (Since right now the remote id is directly reported)

Reply:
That was to cover people using 2024 Elastic, I think it's safe to get rid of the special case for 2027.

Reply:
Which one is the one that Elastic uses in 2025?

Reply:
Elastic

Reply:
I think rather then looking at a list of known dashboards we should just report any NT client ID, so we can get stats on all dashboards and things like limelight, photonvision, etc.